File: Spring Security Tutorials/04th_Custom_Authentication_API_Key/src/main/java/com/example/fourth/config/security/managers/CustomAuthenticationManager.java
package com.example.fourth.config.security.managers;

import com.example.fourth.config.security.providers.APIKeyProvider;
import lombok.AllArgsConstructor;
import org.springframework.security.authentication.AuthenticationManager;
import org.springframework.security.core.Authentication;
import org.springframework.security.core.AuthenticationException;

@AllArgsConstructor
public class CustomAuthenticationManager implements AuthenticationManager {

    private final String key;

    @Override
    public Authentication authenticate(Authentication authentication) throws AuthenticationException {
        var provider = new APIKeyProvider(key);
        if (provider.supports(authentication.getClass())) {
            return provider.authenticate(authentication);
        }

        return authentication;
    }
}

File: Project/IDM/src/main/kotlin/com/project/idm/config/security/managers/CustomAuthenticationManager.kt
//package com.project.idm.config.security.managers
//
//import com.project.idm.config.security.providers.CustomAuthenticationProvider
//import org.springframework.security.authentication.AuthenticationManager
//import org.springframework.security.authentication.BadCredentialsException
//import org.springframework.security.core.Authentication
//import org.springframework.security.core.AuthenticationException
//import org.springframework.stereotype.Component
//
//@Component
//class CustomAuthenticationManager(private val provider: CustomAuthenticationProvider) : AuthenticationManager {
//
////    private lateinit var provider: CustomAuthenticationProvider
////    constructor(provider: CustomAuthenticationProvider) : this() {
////        this.provider = provider
////    }
//
//    @Throws(AuthenticationException::class)
//    override fun authenticate(authentication: Authentication): Authentication {
//        if (provider.supports(authentication.javaClass)) {
//            return provider.authenticate(authentication)
//        }
//        throw BadCredentialsException("Oh No!")
//    }
//}